customising-quickfixj.md

```diff
@@ -0,0 +1,77 @@
+
+### **Incompatible Data Types**
+
+Some incompatable changes have occurred in the evolution of the FIX protocol. For example see below changes to the type of **OrderQty (38)** :
+
+|FIX Version|Field Name|FIX Datatype|Base Type|QuickFIX/J Datatype|
+|---|---|---|---|---|
+|4.0|OrderQty|int|int|```int```|
+|4.2|OrderQty|Qty|float|```Double``` or ```BigDecimal```|
+
+Only one ```quickfix.Field``` class with the same name may be loaded by the Java classloader so only one version of this Field should be in the classpath. QuickFix/J also verifies the data type using the supplied QuickFIX "Dictionary". 
+
+Code generation using ```BigDecimal``` is incompatible at runtime with ```int``` for **OrderQty**. In this case, ```double``` is compatible with ```int``` at run time due to [widening primitive conversion](http://titanium.cs.berkeley.edu/doc/java-langspec-1.0/5.doc.html). 
+
+Runtime incompatabilities can be resolved by:
+* Amending the QuickFIX Dictionary to coerce the code generation and/or validation
+* Changing the ordering of code generation and/or overwrite behaviour of code generation
+* Building independent artefacts for conflicting versions and not using them in the same runtime
+
+See [QuickFIX/J Messages](./quickfixj-messages/readme.md) for details of the build and recommendation for  **how to implement custom builds.**
+
+### **Managing incompatibility with Prior Versions of QuickFIX/J**
+
+From QuickFIX/J 3.0.0 the code generation for ```quickfix.Field``` prefers the FIX Orchestra Standard. This results in incompatible changes to the names of constants. 
+
+For example : ```SettlType.REGULAR_FX_SPOT_SETTLEMENT``` becomes ```SettlType.REGULAR```. 
+
+The required code changes may be trvial in most cases but they are elective. To use ```quickfix-core``` from QuickFIX/J  3.0.0 or later without needing to implement code changes:
+* build the required Message artefacts without the FIX Latest code generation. The Fields will then be generated only from legacy FIX Protocol Versions as they were prior to QuickFIX/J 3.0.0 - **or**
+* if you want to use Messages, Components and/or Fields from FIX Latest while preferring legacy constants, manipulate the order of code generation and/or the over-write behaviour to prefer earlier versions of FIX. For example, prevent FIX Latest code generation from over-writing existing generated sources (Fields).
+
+See [QuickFIX/J Messages](./quickfixj-messages/readme.md) for details of the build and recommendation for  **how to implement custom builds.**
+
+### **Customising the FIX Protocol for specialised Rules of Engagement**
+
+A Rules of Engagement can include customisation Messages, Components and Fields, including User Defined elements.
+
+This can be accomplised by editing the QuickFIX Dictionary or FIX Orchestra Repository (Orchestration) for the specific Rules of Engagement.
```

Comment:
```suggestion
This can be accomplished by editing the QuickFIX Dictionary or FIX Orchestra Repository (Orchestration) for the specific Rules of Engagement.
```